modules/auxiliary/scanner/http/wordpress_scanner.rb

```diff
@@ -10,28 +10,83 @@ class MetasploitModule < Msf::Auxiliary
 
   def initialize
     super(
-      'Name'        => 'Wordpress Scanner',
-      'Description' => 'Detects Wordpress installations and their version number',
-      'Author'      => [ 'Christian Mehlmauer' ],
-      'License'     => MSF_LICENSE
+      'Name' => 'Wordpress Scanner',
+      'Description' => 'Detects Wordpress Versions, Themes, and Plugins',
+      'Author' => [
+        'Christian Mehlmauer', # original module
+        'h00die' # plugins and themes
+      ],
+      'License' => MSF_LICENSE
     )
+    register_options [
+      OptBool.new('THEMES', [false, 'Detect themes', true]),
+      OptBool.new('PLUGINS', [false, 'Detect plugins', true]),
+      OptPath.new('THEMES_FILE', [
+        true, 'File containing themes to enumerate',
+        File.join(Msf::Config.data_directory, 'wordlists', 'wp-themes.txt')
+      ]),
+      OptPath.new('PLUGINS_FILE', [
+        true, 'File containing plugins to enumerate',
+        File.join(Msf::Config.data_directory, 'wordlists', 'wp-plugins.txt')
+      ]),
+      OptInt.new('PROGRESS', [true, 'how often to print progress', 1000])
+    ]
+  end
+
+  def print_progress(i, total)
+    print_status("Progress #{i}/#{total} (#{((i.to_f / total) * 100).truncate(2)}%)")
```

Comment:
If you wanted to make this a little cleaner, you could justify the progress using:

```
    print_status("Progress #{i.to_s.rjust(Math.log10(total).ceil + 1}/#{total} (#{((i.to_f / total) * 100).truncate(2)}%)")
```

That would align the output like:
```
[*] Progress     0/19226 (0.0%)
[*] Progress  1000/19226 (5.2%)
[*] Progress  2000/19226 (10.4%)
[*] Progress  3000/19226 (15.6%)
[*] Progress  4000/19226 (20.8%)
[*] Progress  5000/19226 (26.0%)
[*] Progress  6000/19226 (31.2%)
[*] Progress  7000/19226 (36.4%)
[*] Progress  8000/19226 (41.61%)
[*] Progress  9000/19226 (46.81%)
[*] Progress 10000/19226 (52.01%)
```